File: wagerproof-ios-native/Wagerproof/Features/Props/Detail/PropContextTiles.swift
import SwiftUI
import WagerproofModels
import WagerproofDesign

/// Contextual hit-rate tiles under the chart: L10 (always), day/night split,
/// and (batters only) the vs-archetype split. Mirrors the RN `tiles` block in
/// `PlayerPropDetail`. Low-confidence splits (< 5 games) dim to 0.75 opacity.
struct PropContextTiles: View {
    let row: MLBPlayerPropRow
    let computed: MLBPropComputedAtLine

    private let columns = [
        GridItem(.flexible(), spacing: 8),
        GridItem(.flexible(), spacing: 8),
        GridItem(.flexible(), spacing: 8),
    ]

    var body: some View {
        LazyVGrid(columns: columns, alignment: .leading, spacing: 8) {
            tile(label: "L10", split: computed.l10, subtitle: "Over")

            if let dn = computed.contextualDayNight {
                tile(
                    label: row.gameIsDay ? "☀️ Day" : "🌙 Night",
                    split: dn,
                    subtitle: "Over"
                )
            }

            if let arch = computed.contextualArchetype, let archName = row.oppArchetypeToday {
                tile(
                    label: "vs \(archName) SP",
                    split: arch,
                    subtitle: "Starters"
                )
            }
        }
    }

    private func tile(label: String, split: MLBPropHitSplit, subtitle: String) -> some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(label)
                .font(.system(size: 10, weight: .semibold))
                .tracking(0.5)
                .textCase(.uppercase)
                .foregroundStyle(Color.appTextMuted)
                .lineLimit(1)
                .minimumScaleFactor(0.7)
            Text(split.fractionLabel)
                .font(.system(size: 22, weight: .heavy))
                .foregroundStyle(Color.appPrimary)
                .contentTransition(.numericText())
                .animation(.snappy(duration: 0.3), value: split.fractionLabel)
            Text("\(subtitle) · \(split.pctLabel)")
                .font(.system(size: 10))
                .foregroundStyle(Color.appTextMuted)
                .lineLimit(1)
                .minimumScaleFactor(0.7)
                .contentTransition(.numericText())
                .animation(.snappy(duration: 0.3), value: split.pctLabel)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(10)
        .background(
            RoundedRectangle(cornerRadius: 10, style: .continuous)
                .fill(Color.appSurfaceMuted.opacity(0.6))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 10, style: .continuous)
                .strokeBorder(Color.appBorder, lineWidth: 1)
        )
        .opacity(split.lowConfidence ? 0.75 : 1)
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/props/detail/PropContextTiles.kt
package com.wagerproof.app.features.props.detail

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBPropComputedAtLine
import com.wagerproof.core.models.MLBPropHitSplit
import com.wagerproof.core.models.MLBPlayerPropRow

/**
 * Contextual hit-rate tiles under the chart: L10 (always), day/night split, and
 * (batters only) the vs-archetype split. Low-confidence splits (< 5 games) dim
 * to 0.75. Port of iOS `PropContextTiles.swift`.
 */
@Composable
fun PropContextTiles(row: MLBPlayerPropRow, computed: MLBPropComputedAtLine) {
    val tiles = buildList {
        add(Triple("L10", computed.l10, "Over"))
        computed.contextualDayNight?.let {
            add(Triple(if (row.gameIsDay) "☀️ Day" else "🌙 Night", it, "Over"))
        }
        val arch = computed.contextualArchetype
        val archName = row.oppArchetypeToday
        if (arch != null && archName != null) {
            add(Triple("vs $archName SP", arch, "Starters"))
        }
    }
    // Three-column grid (pad the trailing row so tiles keep equal width).
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        tiles.chunked(3).forEach { rowTiles ->
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.fillMaxWidth()) {
                rowTiles.forEach { (label, split, subtitle) ->
                    ContextTile(label, split, subtitle, Modifier.weight(1f))
                }
                repeat(3 - rowTiles.size) { androidx.compose.foundation.layout.Spacer(Modifier.weight(1f)) }
            }
        }
    }
}

@Composable
private fun ContextTile(label: String, split: MLBPropHitSplit, subtitle: String, modifier: Modifier) {
    val shape = RoundedCornerShape(10.dp)
    Column(
        modifier
            .background(AppColors.appSurfaceMuted.copy(alpha = 0.6f), shape)
            .border(1.dp, AppColors.appBorder, shape)
            .padding(10.dp)
            .alpha(if (split.lowConfidence) 0.75f else 1f),
        verticalArrangement = Arrangement.spacedBy(2.dp),
    ) {
        Text(label.uppercase(), color = AppColors.appTextMuted, fontSize = 10.sp, fontWeight = FontWeight.SemiBold, maxLines = 1)
        Text(split.fractionLabel, color = AppColors.appPrimary, fontSize = 22.sp, fontWeight = FontWeight.Black)
        Text("$subtitle · ${split.pctLabel}", color = AppColors.appTextMuted, fontSize = 10.sp, maxLines = 1)
    }
}
